### ISSUE TITLE
Filter to set number of fragments

### ISSUE THREAD

**Issue by dgnorrod** [10up/ElasticPress#3579]
### Is your enhancement related to a problem? Please describe.

I would like a convenience filter to set the number of fragments for all or any fields.

### Designs

Change [includes/classes/Feature/Search/Search.php](https://github.com/10up/ElasticPress/blob/develop/includes/classes/Feature/Search/Search.php#L236) from:

```
$formatted_args['highlight']['fields'][ $field ] = [
				'pre_tags'            => [ $opening_tag ],
				'post_tags'           => [ $closing_tag ],
				'type'                => 'plain',
				'number_of_fragments' => 0,
			];
```
To:

```
$formatted_args['highlight']['fields'][ $field ] = [
				'pre_tags'            => [ $opening_tag ],
				'post_tags'           => [ $closing_tag ],
				'type'                => 'plain',
				'number_of_fragments' => apply_filters( 'ep_highlight_number_of_fragments', 0, $field ),
			];
```

### Describe alternatives you've considered

I have also considered modifying the formatted args, but this seems easier.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct
```[tasklist]
### Tasks
```

**Comment by felipeelia** [comment]
Hi @dgnorrod, thanks for the suggestion! Do you want to open a PR with that change? If so, don't forget to add the docblock there too, okay? Thanks!

**Comment by dgnorrod** [comment]
I will.

**Comment by dgnorrod** [comment]
I don't see how to branch the repo. Permissions?

**Comment by felipeelia** [comment]
Closed by #3681
